ui/index.d.ts

```diff
@@ -360,6 +362,128 @@ export declare abstract class AzureWizardStep<T> {
     public abstract execute(wizardContext: T, outputChannel: OutputChannel): Promise<T>;
 }
 
+export interface ISubscriptionWizardContext {
+    credentials: ServiceClientCredentials;
+    subscription: Subscription;
+}
+
+export interface ILocationWizardContext extends ISubscriptionWizardContext {
+    /**
+     * You may specify the defaultLocationName if you don't want the `LocationStep` to prompt for a location
+     * For example, if the user selects an existing resource, you might want to use that location as the default for the wizard's other resources
+     */
+    defaultLocationName?: string;
+
+    /**
+     * The location to use for new resources
+     * This value will be defined after `LocationStep.prompt` occurs.
+     */
+    location?: Location;
+}
+
+export declare class LocationStep<T extends ILocationWizardContext> extends AzureWizardStep<T> {
+    public prompt(wizardContext: T, ui: IAzureUserInput): Promise<T>;
+    public execute(wizardContext: T, outputChannel: OutputChannel): Promise<T>;
+}
+
+export interface IAzureNamingRules {
+    minLength: number;
+    maxLength: number;
+
+    /**
+     * A RegExp specifying the invalid characters.
+     * For example, /[^a-z0-9]/ would specify that only lowercase, alphanumeric characters are allowed.
```

Comment:
Don't resource groups have other naming rules?  E.g. I think hyphens are allowed, but not at the end and not two in a row.

Reply:
I believe you're thinking of blob containers: https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names

Here's the docs I found for resource groups:
https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions#naming-rules-and-restrictions